File: Lab3Scripts/LAB3PYTHON/agenda.py
from usuario import Usuario

class Agenda:
    def __init__(self, capacity):
        self._registro = [None] * capacity
        self._no_reg = 0

    def agregar(self, usuario):
        if self.buscar(usuario.get_id()) != -1 or self._no_reg >= len(self._registro):
            return False
        self._registro[self._no_reg] = usuario
        self._no_reg += 1
        return True

    def buscar(self, id):
        for i in range(self._no_reg):
            if self._registro[i].get_id() == id:
                return i
        return -1

    def eliminar(self, id):
        pos = self.buscar(id)
        if pos == -1:
            return False
        for i in range(pos, self._no_reg - 1):
            self._registro[i] = self._registro[i + 1]
        self._registro[self._no_reg - 1] = None
        self._no_reg -= 1
        return True

    def mostrar(self):
        for i in range(self._no_reg):
            print(f"Nombre: {self._registro[i].get_nombre()}, ID: {self._registro[i].get_id()}")

    def mostrar_usuario(self, id):
        pos = self.buscar(id)
        if pos != -1:
            self._registro[pos].mostrar_informacion()
        else:
            print("Usuario no encontrado.")

File: Lab3Scripts/LAB3JAVA/Agenda.java
public class Agenda {
    private Usuario[] registro;
    private int no_reg;

    public Agenda(int capacity) {
        registro = new Usuario[capacity];
        no_reg = 0;
    }

    public boolean agregar(Usuario u) {
        if (buscar(u.getId()) != -1 || no_reg >= registro.length) {
            return false;
        }
        registro[no_reg] = u;
        no_reg++;
        return true;
    }

    public int buscar(int id) {
        for (int i = 0; i < no_reg; i++) {
            if (registro[i].getId() == id) {
                return i;
            }
        }
        return -1;
    }

    public boolean eliminar(int id) {
        int pos = buscar(id);
        if (pos == -1) return false;

        for (int i = pos; i < no_reg - 1; i++) {
            registro[i] = registro[i + 1];
        }
        registro[no_reg - 1] = null;
        no_reg--;
        return true;
    }

    public void mostrar() {
        for (int i = 0; i < no_reg; i++) {
            System.out.println("Nombre: " + registro[i].getNombre() + ", ID: " + registro[i].getId());
        }
    }

    public void mostrarUsuario(int id) {
        int pos = buscar(id);
        if (pos != -1) {
            registro[pos].mostrarInformacion();
        } else {
            System.out.println("Usuario no encontrado.");
        }
    }

    public int getNoReg() {
        return no_reg;
    }
}
